client/my-sites/domains/domain-management/dataviews/use-bulk-action-notice.tsx

```diff
@@ -1,70 +1,125 @@
+import { hasTranslation } from '@wordpress/i18n';
+import { translate, useTranslate } from 'i18n-calypso';
 import { useCallback, useEffect, useMemo, useState } from 'react';
 import { useDispatch } from 'react-redux';
 import wpcomRequest from 'wpcom-proxy-request';
 import { successNotice, errorNotice } from 'calypso/state/notices/actions';
 import { useDomainsDataViewsContext } from './use-context';
 
-type ExistingReactNode = React.ReactElement | string | number;
-type TranslateFunction = (
-	original: string,
-	plural: string,
-	options: TranslateOptionsPlural
-) => ExistingReactNode;
+type TranslateFunction = typeof translate;
 
-const getSuccessMessage = ( job: JobStatus, translate: TranslateFunction ) => {
+const fallbackSuccessMessage = ( job: JobStatus, translate: TranslateFunction ) => {
+	return job.success.length > 1
+		? translate( 'Bulk domain updates finished successfully.' )
+		: translate( 'Domain update finished successfully.' );
+};
+
+const getSuccessMessage = (
+	job: JobStatus,
+	translate: TranslateFunction,
+	isEnglishLocale: boolean
+) => {
 	if ( job.action !== 'set_auto_renew' ) {
-		return translate( 'Your domain has been updated.', 'Your domains have been updated.', {
-			count: job.success.length,
-		} );
+		if ( isEnglishLocale || hasTranslation( 'Your domain has been updated.' ) ) {
+			return translate( 'Your domain has been updated.', 'Your domains have been updated.', {
+				count: job.success.length,
+			} );
+		}
+
+		return fallbackSuccessMessage( job, translate );
 	}
 
 	// If the user tried to enable auto-renew:
 	if ( job.params.auto_renew ) {
+		if (
+			isEnglishLocale ||
+			hasTranslation( 'Automatic renewal has been enabled for your domain.' )
+		) {
+			return translate(
+				'Automatic renewal has been enabled for your domain.',
+				'Automatic renewal has been enabled for your domains.',
+				{ count: job.success.length }
+			);
+		}
+
+		return fallbackSuccessMessage( job, translate );
+	}
+
+	// If the user tried to disable auto-renew:
+	if (
+		isEnglishLocale ||
+		hasTranslation( 'Automatic renewal has been disabled for your domain.' )
+	) {
 		return translate(
-			'Automatic renewal has been enabled for your domain.',
-			'Automatic renewal has been enabled for your domains.',
+			'Automatic renewal has been disabled for your domain.',
+			'Automatic renewal has been disabled for your domains.',
 			{ count: job.success.length }
 		);
 	}
 
-	// If the user tried to disable auto-renew:
-	return translate(
-		'Automatic renewal has been disabled for your domain.',
-		'Automatic renewal has been disabled for your domains.',
-		{ count: job.success.length }
-	);
+	return fallbackSuccessMessage( job, translate );
 };
 
-const getFailureMessage = ( job: JobStatus, translate: TranslateFunction ) => {
+const fallbackFailureMessage = ( job: JobStatus, translate: TranslateFunction ) => {
+	return translate( 'Some domain updates were not successful.' );
```

Comment:
Should we also consider failure count is only one here?

Reply:
For the fallback messages, I used the messages that were here before our changes, so we're sure they are already translated.
There was only this form of the failure message :)

Reply:
Linking to our Slack discussion for references only: p1741155428405409/1741106287.769229-slack-C02NWDZBL0H